notiftime.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * This class manages notifications to TAs after they have helped a student
+ * for a given amount of time. This is to help with remote OH to ping TAs who
+ * have stayed with students for long periods of time.
+ * 
+ * Active TAs are checked each minute after init() has been called to 
+ * check whether they should receive a notification. Because of this, 
+ * notifications may be off by up to 1 minute (although, this should make little 
+ * difference from the perspective of the TA).
+ * 
+ * The proposed alternative was having a timer running for each TA once they
+ * start helping, but that appeared to be more inefficient
+ */
+
+var Sequelize = require('sequelize');
+
+var model = require("./model.js");
+var realtime = require("./realtime.js");
+var options = require("./routes/options.js");
+
+/** Number of milliseconds in a minute */
+const one_min_ms = 60 * 1000;
+
+var helping_tas_cache = null;
+
+exports.clear_helping_tas_cache = function() {
+    helping_tas_cache = null;
+}
+
+exports.init = function() {
+    // Set an interval to run every minute from the time the timer is started
+    setInterval(() => {
+        var now = new Date();
+        
+        // Find all actively helping TAs
+        Sequelize.Promise.props({
+            notif_time_threshold: options.notif_time_threshold(),
+            notif_time_interval: options.notif_time_interval(),
+            tas: function() {
+                if (helping_tas_cache) {
```

Comment:
I've used caches in other places to reduce the load on the database and improve performance for queries that are made on each page load, but I don't think that's necessary here because this only gets run once per minute (and doesn't depend on the number of users).